pkg/controllers/disruption/consolidation.go

```diff
@@ -192,6 +193,49 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 		results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
 	}
 
+	results.NewNodeClaims[0].InstanceTypeOptions = instanceTypesOptionsFilteredByPrice
+	return Command{
+		candidates:   candidates,
+		replacements: results.NewNodeClaims,
+	}, nil
+}
+
+// Allow spot consolidation if SpotConsolidation featureflag is set to true by customers:
+// Fetch all the possible InstanceTypes for a candidate from simulation, sort it by price.
+// Only allow spot consolidation if we have atleast 15 cheapest instance types to replace the current candidate.
+// For single node consolidation, check if the current candidate is part of the first 15 cheapest instance types inorder to avoid repeated consolidation. If yes, do not consolidate.
+// If no, consolidate by sending the filtered list of instanceTypes (instanceTypesOptionsFilteredByPrice) for launch because we want to replace the current candidate with the cheaper instanceType.
+func (c *consolidation) checkForSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate,
+	results *pscheduling.Results, instanceTypesOptionsFilteredByPrice []*cloudprovider.InstanceType) (Command, error) {
+
+	if !options.FromContext(ctx).FeatureGates.SpotConsolidation {
+		if len(candidates) == 1 {
+			// Spot consolidation is turned off.
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "Can't replace a spot node with a spot node")...)
```

Comment:
```suggestion
			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node")...)
```

nit: Makes it more clear what a user needs to do to enable this feature

Reply:
Addressed